libs/agno/agno/os/interfaces/a2a/router.py

```diff
@@ -239,6 +246,244 @@ async def a2a_stream_agent(request: Request, id: str):
+                },
+            },
+            404: {"description": "Agent not found or no tasks found"},
+        },
+    )
+    async def list_agent_tasks(
+        request: Request,
+        id: str,
+        session_type: Optional[SessionType] = None,
+        user_id: Optional[str] = None,
+    ):
+        """List all tasks for an Agent. If session_id is provided in params, returns tasks from that session only."""
+        request_body = await request.json()
+        request_id = request_body.get("id", "unknown")
+        session_id = request_body.get("params", {}).get("session_id")
+
+        agent = get_agent_by_id(id, agents)
+        if not agent:
+            raise HTTPException(status_code=404, detail="Agent not found")
+        db = agent.db
+        if not db:
+            raise HTTPException(status_code=404, detail="Database not configured")
+
+        if hasattr(request.state, "user_id"):
+            user_id = request.state.user_id
+
+        if session_type is None:
+            session_type = SessionType.AGENT
+
+        all_runs = []
+
+        if session_id:
+            # Load specific session
+            if isinstance(db, AsyncBaseDb):
+                session = await db.get_session(
+                    session_id=session_id,
+                    session_type=session_type,
+                    user_id=user_id,
+                    deserialize=False,
+                )
+            else:
+                session = db.get_session(
+                    session_id=session_id,
+                    session_type=session_type,
+                    user_id=user_id,
+                    deserialize=False,
+                )
+
+            if not session:
+                raise HTTPException(status_code=404, detail=f"Session with ID {session_id} not found")
+
+            session_dict = cast(dict, session)
+            runs = session_dict.get("runs", [])
+            all_runs.extend(runs)
+        else:
+            # Load all sessions for this agent
+            if isinstance(db, AsyncBaseDb):
+                sessions, _ = await db.get_sessions(
+                    session_type=session_type,
+                    component_id=id,
+                    user_id=user_id,
+                    deserialize=False,
+                )
+            else:
+                sessions, _ = db.get_sessions(  # type: ignore
+                    session_type=session_type,
+                    component_id=id,
+                    user_id=user_id,
+                    deserialize=False,
+                )
+
+            sessions_list = cast(list, sessions)
+            for session_dict in sessions_list:
+                runs = cast(dict, session_dict).get("runs", [])
+                all_runs.extend(runs)
+
+        # Convert runs to A2A tasks
+        a2a_tasks = [map_run_schema_to_a2a_task(run) for run in all_runs]
+        return {
+            "jsonrpc": "2.0",
+            "id": request_id,
+            "result": a2a_tasks,
+        }
```

Comment:
We are first collecting all runs, then iterating over the full list to map them to a2a tasks. 
We can simplify and just track the tasks:

```python
all_tasks = []

...

all_tasks.extend([map_run_schema_to_a2a_task(run) for run in runs]
```